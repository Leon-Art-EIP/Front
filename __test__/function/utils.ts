import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
