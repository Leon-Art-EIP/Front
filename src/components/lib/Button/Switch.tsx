import { cn } from "../../../tools/cn";

interface ISwitchProps {
  onChange: (checked: boolean) => void;
  isLoading?: boolean;
  isDisabled?: boolean;
  checked: boolean;
  label?: string;
  className?: string;
}

export default function Switch({
  onChange,
  isLoading,
  isDisabled,
  checked,
  label,
  className,
}: ISwitchProps): JSX.Element {
  const handleClick = () => {
    if (!isDisabled && !isLoading) {
      onChange(!checked);
    }
  };

  return (
    <div className={cn("flex flex-col gap-2 items-center justify-center", className)}>
      <span className="text-sm">{label}</span>
      <button
        onClick={handleClick}
        disabled={isDisabled}
        className={cn("relative inline-flex items-center h-6 w-12 rounded-full transition-all", {
          "bg-primary hover:bg-primary-hover": !isDisabled && !isLoading && checked,
          "bg-gray-300": !checked,
          "bg-primary-disabled cursor-not-allowed": isDisabled,
          "animate-pulse": isLoading,
        })}
      >
        <span
          className={cn("absolute left-1 top-1 h-4 w-4 rounded-full transition-transform transform bg-white", {
            "translate-x-6": checked,
            "translate-x-0": !checked,
          })}
        />
      </button>
    </div>
  );
}
