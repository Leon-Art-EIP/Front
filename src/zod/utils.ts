export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
