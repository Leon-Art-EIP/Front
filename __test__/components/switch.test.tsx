import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import Switch from "../../src/components/lib/Button/Switch";

test("Switch snapshot - Checked", () => {
  testSnapshot(<Switch checked={true} onChange={() => {}} label="Switch activé" className="custom-class" />);
});

test("Switch snapshot - Unchecked", () => {
  testSnapshot(<Switch checked={false} onChange={() => {}} label="Switch désactivé" />);
});

test("Switch snapshot - Disabled", () => {
  testSnapshot(<Switch checked={true} onChange={() => {}} isDisabled={true} label="Switch désactivé" />);
});

test("Switch snapshot - Loading", () => {
  testSnapshot(<Switch checked={false} onChange={() => {}} isLoading={true} label="Switch en chargement" />);
});
