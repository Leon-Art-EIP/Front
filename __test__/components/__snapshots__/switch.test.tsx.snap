// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Switch snapshot - Checked 1`] = `
<div
  className="flex flex-col gap-1 items-center justify-center custom-class"
>
  <span
    className="text-sm"
  >
    Switch activé
  </span>
  <button
    className="relative inline-flex items-center h-6 w-12 rounded-full transition-all bg-primary hover:bg-primary-hover"
    onClick={[Function]}
  >
    <span
      className="absolute left-1 top-1 h-4 w-4 rounded-full transition-transform transform bg-white translate-x-6"
    />
  </button>
</div>
`;

exports[`Switch snapshot - Disabled 1`] = `
<div
  className="flex flex-col gap-1 items-center justify-center"
>
  <span
    className="text-sm"
  >
    Switch désactivé
  </span>
  <button
    className="relative inline-flex items-center h-6 w-12 rounded-full transition-all bg-primary-disabled cursor-not-allowed"
    disabled={true}
    onClick={[Function]}
  >
    <span
      className="absolute left-1 top-1 h-4 w-4 rounded-full transition-transform transform bg-white translate-x-6"
    />
  </button>
</div>
`;

exports[`Switch snapshot - Loading 1`] = `
<div
  className="flex flex-col gap-1 items-center justify-center"
>
  <span
    className="text-sm"
  >
    Switch en chargement
  </span>
  <button
    className="relative inline-flex items-center h-6 w-12 rounded-full transition-all bg-gray-300 animate-pulse"
    onClick={[Function]}
  >
    <span
      className="absolute left-1 top-1 h-4 w-4 rounded-full transition-transform transform bg-white translate-x-0"
    />
  </button>
</div>
`;

exports[`Switch snapshot - Unchecked 1`] = `
<div
  className="flex flex-col gap-1 items-center justify-center"
>
  <span
    className="text-sm"
  >
    Switch désactivé
  </span>
  <button
    className="relative inline-flex items-center h-6 w-12 rounded-full transition-all bg-gray-300"
    onClick={[Function]}
  >
    <span
      className="absolute left-1 top-1 h-4 w-4 rounded-full transition-transform transform bg-white translate-x-0"
    />
  </button>
</div>
`;
